src/geo.rs

```diff
@@ -0,0 +1,354 @@
+        Ok(Coord {
+            longitude: values.next().unwrap(),
+            latitude: values.next().unwrap(),
+        })
+    }
+}
+
+impl<T: ToRedisArgs> ToRedisArgs for Coord<T> {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        ToRedisArgs::write_redis_args(&self.longitude, out);
+        ToRedisArgs::write_redis_args(&self.latitude, out);
+    }
+
+    fn is_single_arg(&self) -> bool {
+        false
+    }
+}
+
+/// Options to sort results from [GEORADIUS][1] and [GEORADIUSBYMEMBER][2] commands
+///
+/// [1]: https://redis.io/commands/georadius
+/// [2]: https://redis.io/commands/georadiusbymember
+pub enum RadiusOrder {
+    /// Don't sort the results
+    Unsorted,
+
+    /// Sort returned items from the nearest to the farthest, relative to the center.
+    Asc,
+
+    /// Sort returned items from the farthest to the nearest, relative to the center.
+    Desc,
+}
+
+impl Default for RadiusOrder {
+    fn default() -> RadiusOrder {
+        RadiusOrder::Unsorted
+    }
+}
+
+/// Options for the [GEORADIUS][1] and [GEORADIUSBYMEMBER][2] commands
+///
+/// [1]: https://redis.io/commands/georadius
+/// [2]: https://redis.io/commands/georadiusbymember
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use redis::{Commands, RedisResult};
+/// use redis::geo::{RadiusSearchResult, RadiusOptions, RadiusOrder, Unit};
+/// fn nearest_in_radius(
+///     con: &mut redis::Connection,
+///     key: &str,
+///     longitude: f64,
+///     latitude: f64,
+///     meters: f64,
+///     limit: usize,
+/// ) -> RedisResult<Vec<RadiusSearchResult>> {
+///     let opts = RadiusOptions::default()
+///         .order(RadiusOrder::Asc)
+///         .limit(limit);
+///     con.geo_radius(key, longitude, latitude, meters, Unit::Meters, opts)
+/// }
+/// ```
+#[derive(Default)]
+pub struct RadiusOptions {
+    with_coord: bool,
+    with_dist: bool,
+    count: Option<usize>,
+    order: RadiusOrder,
+    store: Option<Vec<Vec<u8>>>,
+    store_dist: Option<Vec<Vec<u8>>>,
+}
+
+impl RadiusOptions {
+    /// Limit the results to the first N matching items.
+    pub fn limit(mut self, n: usize) -> Self {
+        self.count = Some(n);
+        self
+    }
+
+    /// Return the distance of the returned items from the specified center.
+    /// The distance is returned in the same unit as the unit specified as the
+    /// radius argument of the command.
+    pub fn with_dist(mut self) -> Self {
+        self.with_dist = true;
+        self
+    }
+
+    /// Return the longitude,latitude coordinates of the matching items.
+    pub fn with_coord(mut self) -> Self {
+        self.with_coord = true;
+        self
+    }
+
+    /// Sort the returned items
+    pub fn order(mut self, o: RadiusOrder) -> Self {
+        self.order = o;
+        self
+    }
+
+    /// Store the results in a sorted set at key, instead of returning them.
+    ///
+    /// This feature can't be used with any `with_*` method.
```

Comment:
_Maybe_ we can later add some way to verify that at runtime (or statically? but that requires a bit more code, not too much a fan of)